test/e2e/scripts/e2e-ci.sh

```
@@ -1,31 +1,359 @@
+    if [[ "$E2E_CLEANUP" != "true" ]]; then
+        echo "Skipping cleanup (E2E_CLEANUP=$E2E_CLEANUP)"
+        return
+    fi
+    echo ""
+    echo "=== Cleanup ==="
+    kubectl delete job "$E2E_JOB_NAME" -n "$E2E_NS" --ignore-not-found 2>/dev/null || true
+    helm uninstall "$HELM_RELEASE" -n "$GATEWAY_NAMESPACE" 2>/dev/null || true
+    kubectl delete namespace "$E2E_NS" --ignore-not-found --timeout=60s 2>/dev/null || true
+    kubectl delete gateway "$GATEWAY_NAME" -n "$GATEWAY_NAMESPACE" --ignore-not-found 2>/dev/null || true
+    echo "Cleanup complete"
+}
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)